iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/serviceclient/ServiceClientTests.java

```diff
@@ -75,9 +85,42 @@ public static Collection inputsCommon() throws IOException
         return inputs;
     }
 
+    @BeforeClass
+    public static void startProxy()
+    {
+        proxyServer = DefaultHttpProxyServer.bootstrap()
+                        .withPort(testProxyPort)
+                        .start();
+    }
+
+
+    @AfterClass
+    public static void stopProxy()
+    {
+        proxyServer.stop();
+    }
+
     @Test (timeout=MAX_TEST_MILLISECONDS)
     @ConditionalIgnoreRule.ConditionalIgnore(condition = StandardTierOnlyRule.class)
     public void cloudToDeviceTelemetry() throws Exception
+    {
+        cloudToDeviceTelemetry(false);
+    }
+
+    @Test (timeout=MAX_TEST_MILLISECONDS)
+    @ConditionalIgnoreRule.ConditionalIgnore(condition = StandardTierOnlyRule.class)
+    public void cloudToDeviceTelemetryWithProxy() throws Exception
+    {
+        if (testInstance.protocol != IotHubServiceClientProtocol.AMQPS_WS)
+        {
+            //Proxy support only exists for AMQPS_WS currently
+            return;
```

Comment:
What are the hurdles with AMQP?

Reply:
It may be possible, but no one has asked for this yet. The customer I'm writing this for just wants AMQPS_WS to support tunneling through HTTP proxies